### PR TITLE
[3.x] added optional cropping for texture atlas importer

### PR DESCRIPTION
**Pull request by boruok**
added optional region cropping to Texture Atlas importer. See godotengine/godot-proposals#1661

this is my second try of #52564, sorry, first ever PR. Let me know if did anything wrong.

### CODE REVIEW

File: editor/import/resource_importer_texture_atlas.cpp [hunk 1]
```diff
@@ -71,6 +71,7 @@ String ResourceImporterTextureAtlas::get_preset_name(int p_idx) const {
 void ResourceImporterTextureAtlas::get_import_options(List<ImportOption> *r_options, int p_preset) const {
 	r_options->push_back(ImportOption(PropertyInfo(Variant::STRING, "atlas_file", PROPERTY_HINT_SAVE_FILE, "*.png"), ""));
 	r_options->push_back(ImportOption(PropertyInfo(Variant::INT, "import_mode", PROPERTY_HINT_ENUM, "Region,Mesh2D"), 0));
+	r_options->push_back(ImportOption(PropertyInfo(Variant::BOOL, "cropped"), false));
```
Comment (akien-mga) [inline editor/import/resource_importer_texture_atlas.cpp:74]: Would it be clearer if the option was named `crop_to_region` or something like that?

Comment (boruok) [reply editor/import/resource_importer_texture_atlas.cpp]: done

**Comment by akien-mga** [comment]
Code looks good to me, I just need to get the feature itself validated by import folks and we can merge.

This should probably have a `master` version too.

**Comment by fire** [comment]
The proposal describes a real use. Will wait for the master branch pr.

**Review by reduz — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

Merging already for `3.x` to include it in 3.4 beta 6, but please remember to make a PR for `master` too so that both branches have the feature.

**Pull request merged**